src/Diagnostics.RuntimeHost/Controllers/Site/SitesController.cs

```diff
@@ -76,7 +80,50 @@ public SitesController(IStampService stampService, ISiteService siteService, ICo
             
             App app = await GetAppResource(subscriptionId, resourceGroupName, siteName, postBody, startTimeUtc, endTimeUtc);
             OperationContext<App> context = PrepareContext<App>(app, startTimeUtc, endTimeUtc);
-            return await base.GetDetector<App>(detectorId, context);
+            return await base.GetDetectorResponse<App>(detectorId, context);
+        }
+
+        [HttpPost(UriElements.Insights)]
+        public async Task<IActionResult> GetInsights(string subscriptionId, string resourceGroupName, string siteName, [FromBody] DiagnosticSiteData postBody, string supportTopicId, string startTime = null, string endTime = null, string timeGrain = null)
+        {
+            if (postBody == null)
+            {
+                return BadRequest("Post Body missing.");
+            }
+
+            if (siteName == "steve")
+            {
+                subscriptionId = "ef90e930-9d7f-4a60-8a99-748e0eea69de";
+                resourceGroupName = "mikono-demo";
+                siteName = "NodeBakery";
+                postBody.Name = "NodeBakery";
+                postBody.IsLinux = true;
+                postBody.Stamp = new DiagnosticStampData()
+                {
+                    Name = "waws-prod-bay-081",
+                    InternalName = "waws-prod-bay-081"
+                };
+                postBody.HostNames = new List<DiagnosticHostnameData>()
+                {
+                    new DiagnosticHostnameData()
+                    {
+                        Name = "nodebakery.azurewebsites.net",
+                        Type = 0
+                    }
+                };
+            }
+
+            DateTimeHelper.PrepareStartEndTimeWithTimeGrain(string.Empty, string.Empty, string.Empty, out DateTime startTimeUtc, out DateTime endTimeUtc, out TimeSpan timeGrainTimeSpan, out string errorMessage);
+            App app = await GetAppResource(subscriptionId, resourceGroupName, siteName, postBody, startTimeUtc, endTimeUtc);
```

Comment:
Shouldn't you move this detector to Base and have this Controller inherit that.  Check any other API above.
Its a good idea to keep common things there in base. It will help us onboard new resources quickly.